frontend/public/components/modals/impersonate-user-modal.tsx

```diff
@@ -0,0 +1,357 @@
+      return availableGroups;
+    }
+    return availableGroups.filter((group) =>
+      group.toLowerCase().includes(groupSearchFilter.toLowerCase()),
+    );
+  }, [groupSearchFilter, availableGroups]);
+
+  const textInputGroupRef = React.useRef<HTMLDivElement>(null);
+
+  const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+    <MenuToggle
+      ref={toggleRef}
+      variant="typeahead"
+      onClick={() => setIsGroupSelectOpen(!isGroupSelectOpen)}
+      isExpanded={isGroupSelectOpen}
+      style={{ width: '100%' }}
+    >
+      <TextInputGroup isPlain>
+        <TextInputGroupMain
+          value={groupSearchFilter}
+          onClick={() => setIsGroupSelectOpen(!isGroupSelectOpen)}
+          onChange={(_event, value) => {
+            setGroupSearchFilter(value);
+            if (!isGroupSelectOpen) {
+              setIsGroupSelectOpen(true);
+            }
+          }}
+          autoComplete="off"
+          innerRef={textInputGroupRef}
+          placeholder={t('public~Enter groups')}
+          role="combobox"
+          isExpanded={isGroupSelectOpen}
+          aria-controls="select-typeahead-listbox"
+        />
+        <TextInputGroupUtilities>
+          {groupSearchFilter && (
+            <Button
+              variant="plain"
+              onClick={() => {
+                setGroupSearchFilter('');
+                textInputGroupRef?.current?.focus();
+              }}
+              aria-label="Clear input"
+            >
+              <TimesIcon aria-hidden />
+            </Button>
+          )}
+        </TextInputGroupUtilities>
+      </TextInputGroup>
+    </MenuToggle>
+  );
+
+  return (
+    <Modal
+      variant={ModalVariant.small}
+      title={t('public~Impersonate user')}
+      isOpen={isOpen}
+      onClose={handleClose}
+      actions={[
+        <Button
+          key="impersonate"
+          variant="primary"
+          onClick={handleImpersonate}
+          data-test="impersonate-button"
+        >
+          {t('public~Impersonate')}
+        </Button>,
+        <Button key="cancel" variant="link" onClick={handleClose} data-test="cancel-button">
+          {t('public~Cancel')}
+        </Button>,
+      ]}
+    >
+      <Form>
+        <Alert
+          variant={AlertVariant.warning}
+          isInline
+          title={t('public~Impersonating a user or group grants you their exact permissions.')}
+        />
+
+        <FormGroup
+          label={
+            <span style={{ display: 'inline-flex', alignItems: 'center', gap: '0.25rem' }}>
+              {t('public~Username')}
+              <Popover
+                headerContent={t('public~Username')}
+                bodyContent={t('public~The name of the user to impersonate')}
+              >
+                <button
+                  type="button"
+                  aria-label={t('public~More info for username field')}
+                  onClick={(e) => e.preventDefault()}
+                  style={{
```

Comment:
inline styles to this extent are discouraged, try to find a pf component which solves your problem